org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -703,6 +713,19 @@ public Preferences setImportGradleEnabled(boolean enabled) {
 		return this;
 	}
 
+	@SuppressWarnings("restriction")
+	public Preferences setImportGradleOfflineEnabled(boolean enabled) {
```

Comment:
I'd rather prefer we decouple preferences from Gradle (or any build system). We should have preference listeners for that